doc/specs/#166 - Font Support.md

```diff
@@ -0,0 +1,416 @@
+
+In addition, a new registry entry is created in the following registry paths:
+The name of the registry key typically matches the title of the font file, and the value is set to the path of the font file.
+
+#### Local Machine Registry
+---
+
+`Computer\HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Fonts`
+
+| Name    | Type | Value |
+| -------- | ------- | ------- |
+| Calibri Bold Italic (TrueType)  | REG_SZ    | calibriz.ttf |
+
+> Note that for the LOCAL_MACHINE registry entry, only the file name is specified, not the full path.
+
+#### Current User Registry
+---
+
+ `Computer\HKEY_CURRENT_USER\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Fonts`
+
+| Name    | Type | Value |
+| -------- | ------- | ------- |
+| Roboto Bold Italic (TrueType)  | REG_SZ    | C:\Users\ryfu\AppData\Local\Microsoft\Windows\Fonts\Roboto-BoldItalic.ttf |
+
+
+### Uninstalling Fonts
+To remove a font, the inverse operation of the steps above need to be completed. 
+1. Remove the registry key entry for that specific font file
+2. Remove the file from the font directory
+
+### Relevant APIS:
+- [DWriteCreateFactory](https://learn.microsoft.com/en-us/windows/win32/api/dwrite/nf-dwrite-dwritecreatefactory) - Creates the DirectWrite factory object for calling DWrite apis
+- [IDWriteFactory::GetSystemFontCollection](https://learn.microsoft.com/en-us/windows/win32/api/dwrite/nf-dwrite-idwritefactory-getsystemfontcollection) - Obtains the installed font collections
+
+- [IDWriteFontCollection::FindFamilyName](https://learn.microsoft.com/en-us/windows/win32/api/dwrite/nf-dwrite-idwritefontcollection-findfamilyname) - Finds a font family with a specified family name. Returns the index number in the font collection.
+- [IDWriteFontCollection::GetFontFamily](https://learn.microsoft.com/en-us/windows/win32/api/dwrite/nf-dwrite-idwritefontcollection-getfontfamily) - Creates the font family object using a provided index. Must be a valid index from the font collection.
+- [IDWriteFontList::GetFont](https://learn.microsoft.com/en-us/windows/win32/api/dwrite/nf-dwrite-idwritefontlist-getfont) - Gets the font from a family with a zero-based index
+- [IDWriteFont::GetInformationalStrings method](https://learn.microsoft.com/en-us/windows/win32/api/dwrite/nf-dwrite-idwritefont-getinformationalstrings) - Gets specific informational strings such as font full name, author, etc.
+- [IDWriteLocalFontFileLoader::GetFilePathFromKey](https://learn.microsoft.com/en-us/windows/win32/api/dwrite/nf-dwrite-idwritelocalfontfileloader-getfilepathfromkey) - Obtains the absolute font file path from a font file reference key.
+- [SHGetPropertyStoreFromParsingName](https://learn.microsoft.com/en-us/windows/win32/api/shobjidl_core/nf-shobjidl_core-shgetpropertystorefromparsingname) - Returns the property store for a given path to obtain details like the file's title.
+
+
+## Manifest Changes
+- Addition of `font` to `installerType` and `nestedInstallerType`
```

Comment:
Is there any way for the manifest to specify the font name and not just the path to it? Thinking of a use case where the file is something like `ARandomFont.ttf` when it the name of the font is actually something like `Really Awesome Font`

Reply:
I believe that the font files contain face and family metadata already and we are planning to always use that.  If that is not always the case, then we do probably need a way to specify the name.  The registry value names don't appear to have any actual impact on the enumerated fonts (from what I remember of @ryfu-msft investigation), but maybe they are important in the event that the file doesn't contain the relevant metadata.